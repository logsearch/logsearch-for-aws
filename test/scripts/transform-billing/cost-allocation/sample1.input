Don't see your tags in the report? New tags are excluded by default - go to https://console.aws.amazon.com/billing/home#/preferences/tags to update your cost allocation keys.
"InvoiceID","PayerAccountId","LinkedAccountId","RecordType","RecordID","BillingPeriodStartDate","BillingPeriodEndDate","InvoiceDate","PayerAccountName","LinkedAccountName","TaxationAddress","PayerPONumber","ProductCode","ProductName","SellerOfRecord","UsageType","Operation","AvailabilityZone","RateId","ItemDescription","UsageStartDate","UsageEndDate","UsageQuantity","BlendedRate","CurrencyCode","CostBeforeTax","Credits","TaxAmount","TaxType","TotalCost","user:deployment","user:director"
"12345678","123456789012","123456789012","PayerLineItem","3000000000000000001","2015/07/01 00:00:00","2015/07/31 23:59:59","2015/08/03 11:51:47","John Doe","","123 Main St, San Jose, California, 95131, US",,"AWSDataTransfer","AWS Data Transfer","Amazon Web Services, Inc.","USW1-USE1-AWS-Out-Bytes",,"","1234567","$0.02 per GB - US West (Northern California) data transfer to US East (Northern Virginia)","2015/07/01 00:00:00","2015/07/31 23:59:59","0.00001191",,"USD","0.000000","0.000000","0.000000","None","0.000000",,
"12345678","123456789012","123456789012","PayerLineItem","3000000000000000002-0","2015/07/01 00:00:00","2015/07/31 23:59:59","2015/08/03 11:51:47","John Doe","","123 Main St, San Jose, California, 95131, US",,"AWSDataTransfer","AWS Data Transfer","Amazon Web Services, Inc.","USW2-APS1-AWS-Out-Bytes",,"","1234568","$0.02 per GB - US West (Oregon) data transfer to Asia Pacific (Singapore)","2015/07/01 00:00:00","2015/07/31 23:59:59","0.00001785",,"USD","0.000000","0.000000","0.000000","None","0.000000",,
"12345678","123456789012","123456789012","PayerLineItem","3000000000000000002-1","2015/07/01 00:00:00","2015/07/31 23:59:59","2015/08/03 11:51:47","John Doe","","123 Main St, San Jose, California, 95131, US",,"AWSDataTransfer","AWS Data Transfer","Amazon Web Services, Inc.","USW2-APS1-AWS-Out-Bytes",,"","1234569","$0.02 per GB - US West (Oregon) data transfer to Asia Pacific (Singapore)","2015/07/01 00:00:00","2015/07/31 23:59:59","0.72922732",,"USD","0.010000","0.000000","0.000000","None","0.010000","frontend-r0","prod-abraxas-aws-usw2"
"12345678","123456789012","123456789012","PayerLineItem","3000000000000000003","2015/07/01 00:00:00","2015/07/31 23:59:59","2015/08/03 11:51:47","John Doe","","123 Main St, San Jose, California, 95131, US",,"AmazonEC2","Amazon Elastic Compute Cloud","Amazon Web Services, Inc.","EUC1-EBS:VolumeUsage",,"","1234570","$0.059 per GB-month of Magnetic provisioned storage - EU (Frankfurt)","2015/07/01 00:00:00","2015/07/19 05:29:40","7.90591317",,"USD","0.470000","0.000000","0.000000","None","0.470000",,
