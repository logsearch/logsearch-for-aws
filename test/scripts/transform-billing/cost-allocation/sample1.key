123456789012-aws-cost-allocation-2014-07.csv
