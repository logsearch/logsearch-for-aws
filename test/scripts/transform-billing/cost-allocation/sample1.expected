{"InvoiceID":12345678,"PayerAccountId":123456789012,"LinkedAccountId":123456789012,"RecordType":"PayerLineItem","RecordID":"3000000000000000001","BillingPeriodStartDate":"2015/07/01 00:00:00","BillingPeriodEndDate":"2015/07/31 23:59:59","InvoiceDate":"2015/08/03 11:51:47","PayerAccountName":"John Doe","LinkedAccountName":null,"TaxationAddress":"123 Main St, San Jose, California, 95131, US","PayerPONumber":null,"ProductCode":"AWSDataTransfer","ProductName":"AWS Data Transfer","SellerOfRecord":"Amazon Web Services, Inc.","UsageType":"USW1-USE1-AWS-Out-Bytes","Operation":null,"AvailabilityZone":null,"RateId":1234567,"ItemDescription":"$0.02 per GB - US West (Northern California) data transfer to US East (Northern Virginia)","UsageStartDate":"2015/07/01 00:00:00","UsageEndDate":"2015/07/31 23:59:59","UsageQuantity":1.191e-05,"BlendedRate":null,"CurrencyCode":"USD","CostBeforeTax":0.0,"Credits":0.0,"TaxAmount":0.0,"TaxType":"None","TotalCost":0.0,"@source":{"aws_file":"123456789012-aws-cost-allocation-2014-07.csv","aws_type":"cost-allocation"},"SubscriptionId":0,"PricingPlanId":0,"Rate":0.0,"Cost":0.0,"user":{"deployment":null,"director":null}}
{"InvoiceID":12345678,"PayerAccountId":123456789012,"LinkedAccountId":123456789012,"RecordType":"PayerLineItem","RecordID":"3000000000000000002-0","BillingPeriodStartDate":"2015/07/01 00:00:00","BillingPeriodEndDate":"2015/07/31 23:59:59","InvoiceDate":"2015/08/03 11:51:47","PayerAccountName":"John Doe","LinkedAccountName":null,"TaxationAddress":"123 Main St, San Jose, California, 95131, US","PayerPONumber":null,"ProductCode":"AWSDataTransfer","ProductName":"AWS Data Transfer","SellerOfRecord":"Amazon Web Services, Inc.","UsageType":"USW2-APS1-AWS-Out-Bytes","Operation":null,"AvailabilityZone":null,"RateId":1234568,"ItemDescription":"$0.02 per GB - US West (Oregon) data transfer to Asia Pacific (Singapore)","UsageStartDate":"2015/07/01 00:00:00","UsageEndDate":"2015/07/31 23:59:59","UsageQuantity":1.785e-05,"BlendedRate":null,"CurrencyCode":"USD","CostBeforeTax":0.0,"Credits":0.0,"TaxAmount":0.0,"TaxType":"None","TotalCost":0.0,"@source":{"aws_file":"123456789012-aws-cost-allocation-2014-07.csv","aws_type":"cost-allocation"},"SubscriptionId":0,"PricingPlanId":0,"Rate":0.0,"Cost":0.0,"user":{"deployment":null,"director":null}}
{"InvoiceID":12345678,"PayerAccountId":123456789012,"LinkedAccountId":123456789012,"RecordType":"PayerLineItem","RecordID":"3000000000000000002-1","BillingPeriodStartDate":"2015/07/01 00:00:00","BillingPeriodEndDate":"2015/07/31 23:59:59","InvoiceDate":"2015/08/03 11:51:47","PayerAccountName":"John Doe","LinkedAccountName":null,"TaxationAddress":"123 Main St, San Jose, California, 95131, US","PayerPONumber":null,"ProductCode":"AWSDataTransfer","ProductName":"AWS Data Transfer","SellerOfRecord":"Amazon Web Services, Inc.","UsageType":"USW2-APS1-AWS-Out-Bytes","Operation":null,"AvailabilityZone":null,"RateId":1234569,"ItemDescription":"$0.02 per GB - US West (Oregon) data transfer to Asia Pacific (Singapore)","UsageStartDate":"2015/07/01 00:00:00","UsageEndDate":"2015/07/31 23:59:59","UsageQuantity":0.72922732,"BlendedRate":null,"CurrencyCode":"USD","CostBeforeTax":0.01,"Credits":0.0,"TaxAmount":0.0,"TaxType":"None","TotalCost":0.01,"@source":{"aws_file":"123456789012-aws-cost-allocation-2014-07.csv","aws_type":"cost-allocation"},"SubscriptionId":0,"PricingPlanId":0,"Rate":0.0,"Cost":0.0,"user":{"deployment":"frontend-r0","director":"prod-abraxas-aws-usw2"}}
{"InvoiceID":12345678,"PayerAccountId":123456789012,"LinkedAccountId":123456789012,"RecordType":"PayerLineItem","RecordID":"3000000000000000003","BillingPeriodStartDate":"2015/07/01 00:00:00","BillingPeriodEndDate":"2015/07/31 23:59:59","InvoiceDate":"2015/08/03 11:51:47","PayerAccountName":"John Doe","LinkedAccountName":null,"TaxationAddress":"123 Main St, San Jose, California, 95131, US","PayerPONumber":null,"ProductCode":"AmazonEC2","ProductName":"Amazon Elastic Compute Cloud","SellerOfRecord":"Amazon Web Services, Inc.","UsageType":"EUC1-EBS:VolumeUsage","Operation":null,"AvailabilityZone":null,"RateId":1234570,"ItemDescription":"$0.059 per GB-month of Magnetic provisioned storage - EU (Frankfurt)","UsageStartDate":"2015/07/01 00:00:00","UsageEndDate":"2015/07/19 05:29:40","UsageQuantity":7.90591317,"BlendedRate":null,"CurrencyCode":"USD","CostBeforeTax":0.47,"Credits":0.0,"TaxAmount":0.0,"TaxType":"None","TotalCost":0.47,"@source":{"aws_file":"123456789012-aws-cost-allocation-2014-07.csv","aws_type":"cost-allocation"},"SubscriptionId":0,"PricingPlanId":0,"Rate":0.0,"Cost":0.0,"user":{"deployment":null,"director":null}}
